js/kucoin.js

```diff
@@ -1,1452 +1,525 @@
-        let orders = response['data']['datas'];
-        //
-        // the caching part to be removed
-        //
-        //     for (let i = 0; i < orders.length; i++) {
-        //         let order = this.parseOrder (this.extend (orders[i], {
-        //             'status': 'closed',
-        //         }), market);
-        //         let orderId = order['id'];
-        //         if (orderId in this.orders)
-        //             if (this.orders[orderId]['status'] === 'canceled')
-        //                 order['status'] = this.orders[orderId]['status'];
-        //         this.orders[order['id']] = order;
-        //     }
-        //     let closedOrders = this.filterBy (this.orders, 'status', 'closed');
-        //     return this.filterBySymbolSinceLimit (closedOrders, symbol, since, limit);
-        //
-        return this.parseOrdersByStatus (orders, market, since, limit, 'closed');
+        const currencyId = this.currencyId (code);
+        const request = { 'currency': currencyId };
+        return await this.privateGetDepositAddresses (this.extend (request, params));
     }
 
-    priceToPrecision (symbol, price) {
-        const market = this.market (symbol);
-        const code = market['quote'];
-        return this.decimalToPrecision (price, ROUND, this.currencies[code]['precision'], this.precisionMode);
-    }
-
-    amountToPrecision (symbol, amount) {
-        const market = this.market (symbol);
-        const code = market['base'];
-        return this.decimalToPrecision (amount, TRUNCATE, this.currencies[code]['precision'], this.precisionMode);
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const marketId = this.marketId (symbol);
+        const request = { 'symbol': marketId, 'level': 3 };
+        const response = await this.publicGetMarketOrderbookLevelLevel (this.extend (request, params));
+        // { sequence: '1547731421688',
+        //   asks: [ [ '5c419328ef83c75456bd615c', '0.9', '0.09' ], ... ],
+        //   bids: [ [ '5c419328ef83c75456bd615c', '0.9', '0.09' ], ... ], }
+        const responseData = response['data'];
+        const timestamp = this.safeInteger (responseData, 'sequence');
+        return this.parseOrderBook (responseData, timestamp, 'bids', 'asks', 1, 2);
     }
 
     async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
-        if (type !== 'limit')
-            throw new ExchangeError (this.id + ' allows limit orders only');
         await this.loadMarkets ();
-        let market = this.market (symbol);
-        let request = {
-            'symbol': market['id'],
-            'type': side.toUpperCase (),
-            'price': this.priceToPrecision (symbol, price),
-            'amount': this.amountToPrecision (symbol, amount),
+        const marketId = this.marketId (symbol);
+        // required param, cannot be used twice
+        const clientOid = String (this.random ());
```

Comment:
I suggest that we drop `random` and use the already-existing `.uuid()` function for user-defined order ids throughout this file. How does that sound to you?